lib/app/view/app.dart

```diff
@@ -1,55 +1,34 @@
+import 'package:black_tax_and_white_benefits/app/data/post_client.dart';
+import 'package:black_tax_and_white_benefits/app/view/post_cell.dart';
 import 'package:flutter/material.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
 
-class App extends StatelessWidget {
+class App extends ConsumerWidget {
   const App({super.key});
 
   @override
-  Widget build(BuildContext context) => const MaterialApp(
-        title: 'Counter App',
-        home: MyHomePage(title: 'Counter App Home Page'),
-      );
-}
-
-class MyHomePage extends StatefulWidget {
-  const MyHomePage({required this.title, super.key});
-
-  final String title;
-
-  @override
-  State<MyHomePage> createState() => _MyHomePageState();
-}
-
-class _MyHomePageState extends State<MyHomePage> {
-  int _counter = 0;
+  Widget build(BuildContext context, WidgetRef ref) {
+    final posts = ref.watch(getPostsProvider);
 
-  void _incrementCounter() => setState(() => _counter++);
+    debugPrint(posts.valueOrNull?.map((e) => '${e.id}').toString());
 
-  @override
-  Widget build(BuildContext context) => Scaffold(
-        appBar: AppBar(
-          title: Text(widget.title),
-        ),
+    return MaterialApp(
+      home: Scaffold(
+        appBar: AppBar(title: const Text('Black Tax White Benefits')),
         body: Center(
-          child: Column(
-            mainAxisAlignment: MainAxisAlignment.center,
-            children: <Widget>[
-              const Text(
-                'Someone have tapped the button this many times:',
-              ),
-              Text(
-                '$_counter',
-                style: Theme.of(context).textTheme.headlineMedium,
-              ),
-            ],
+          child: posts.when(
```

Comment:
@alexdivadi  plz make a jira ticket for us to make this reusable code either with defaultWhen extension or codewithandrea's AsyncValueWidget or whatever it's called

Reply:
Linked here: [WA-24](https://anirac-tech.atlassian.net/browse/WA-24?atlOrigin=eyJpIjoiODQwMjk3YTE3MDQ5NGU5NDgzYmQ0YjhkMDNjOWNlMGMiLCJwIjoiaiJ9)